server/chat-commands/core.ts

```diff
@@ -642,7 +642,58 @@ export const commands: ChatCommands = {
 		`/busy OR /donotdisturb - Marks you as busy.`,
 		`Use /donotdisturb to also block private messages and challenges.`,
 		`Use /back to mark yourself as back.`,
-	 ],
+	],
+
+	pmbl: 'pmblocklist',
+	pmblocklist: {
+		add(target, room, user) {
+			// hotpatching
+			Chat.updateSettings(user);
+
+			target = toID(target);
+			if (!target) return this.parse('/pmblocklist');
+			if (target === user.id) return this.errorReply(`You cannot block yourself.`);
+			if (!user.settings.blockPMs.specific) user.settings.blockPMs.specific = [];
+			const specific = user.settings.blockPMs.specific;
+			if (specific.includes(target)) {
+				return this.errorReply(`You are already blocking the user "${target}" from PMing you.`);
+			}
+			specific.push(target);
+			user.update();
+			this.sendReply(`The user "${target}" can not longer PM you, unless they're global staff.`);
+			return this.parse(`/pmblocklist view`);
+		},
+		remove(target, room, user) {
+			Chat.updateSettings(user);
+
+			target = toID(target);
+			if (!target) return this.parse('/pmblocklists');
+			if (!user.settings.blockPMs.specific) user.settings.blockPMs.specific = [];
+			const specific = user.settings.blockPMs.specific;
+			const index = specific.indexOf(target);
+			if (index < 0) return this.errorReply(`The user "${target}" is not on your PM blocklist.`);
+			specific.splice(index, 1);
+			user.update();
+			this.sendReply(`You have removed the user "${target}" from your PM blocklist.`);
+			return this.parse(`/pmblocklist view`);
+		},
+		view(target, room, user) {
+			Chat.updateSettings(user);
+
+			if (!user.settings.blockPMs.specific) user.settings.blockPMs.specific = [];
```

Comment:
Pretty sure this will mean everyones type looks like `{all: ..., specific: []}` anyway as things converge - nothing removes `specific: []` AFAICT, so once this is ever assigned on a user object it will pretty much last forever (because the client will persist it and propagate it across the wire.

The point being - might as well just stop making `specific` optional because in practice it will always exist eventually anyway